### PR TITLE
fix(kad): prevent simultaneous dials to peer

### PR DESCRIPTION
**Pull request by b-zee**
## Description

By default, dialing is prevented when already connected. `kad` already checks this manually before doing dials. By changing the condition from `Disconnected` to `NotDialing`, we prevent simultaneous dials to a peer.

## Change checklist

- [ ] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: protocols/kad/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,10 @@
+## 0.44.3
```
Comment (thomaseizinger) [inline protocols/kad/CHANGELOG.md]:
```suggestion
## 0.44.3 - unreleased
```

File: protocols/kad/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,10 @@
...
+
+- Prevent simultaneous dials to peer.
```
Comment (thomaseizinger) [inline protocols/kad/CHANGELOG.md]:
```suggestion
- Prevent simultaneous dials to peers.
```

File: protocols/kad/CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,10 @@
...
+  See [PR 4224]
```
Comment (thomaseizinger) [inline protocols/kad/CHANGELOG.md]:
```suggestion
  See [PR 4224].
```

File: protocols/kad/Cargo.toml [hunk 4]
```diff
@@ -3,7 +3,7 @@ name = "libp2p-kad"
 edition = "2021"
 rust-version = "1.65.0"
 description = "Kademlia protocol for libp2p"
-version = "0.44.2"
+version = "0.44.3"
```
Comment (thomaseizinger) [inline protocols/kad/Cargo.toml:6]: You will also need to bump the version in the root manifest in our `workspace.dependencies`.

**Review by thomaseizinger — APPROVED** [review]
Ready from my end :)

File: protocols/kad/CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,10 @@
+## 0.44.3 - unreleased
```
Comment (mxinden) [inline protocols/kad/CHANGELOG.md:1]:
```suggestion
## 0.44.3
```

Adding `- unreleased` is the way to go. Thank you. Though in this case I will cut a release right away.

**Review by mxinden — APPROVED** [review]
Thank you for bearing with us!

**Comment by mxinden** [comment]
[Published](https://crates.io/crates/libp2p-kad/0.44.3) and [tagged](https://github.com/libp2p/rust-libp2p/releases/tag/libp2p-kad-v0.44.3).

//CC @b-zee

**Pull request merged**